packages/convex-helpers/server/pagination.ts

```diff
@@ -0,0 +1,301 @@
+import { Value, convexToJson } from "convex/values";
+import {
+  DocumentByName,
+  GenericDataModel,
+  GenericDatabaseReader,
+  IndexNames,
+  NamedTableInfo,
+  SchemaDefinition,
+  TableNamesInDataModel,
+} from "convex/server";
+
+export type IndexKey = Value[];
+
+export type PageRequest<
+  DataModel extends GenericDataModel,
+  T extends TableNamesInDataModel<DataModel>,
+> = {
+  /// Request a page of documents from this table.
+  table: T;
+  /// Where the page starts. Default or empty array is the start of the table.
+  startIndexKey?: IndexKey;
```

Comment:
a future types rabbit hole to go down would be to type this based on a prefix of the `index` from `schema` specified or defaulting to a prefix of `[ number, GenericId<T> ]` unless they manually specify `indexFields`